src/main/scala/prci/ClockBundles.scala

```diff
@@ -12,5 +14,8 @@ class ClockBundle(val params: ClockBundleParameters) extends Bundle
 
 class ClockGroupBundle(val params: ClockGroupBundleParameters) extends Bundle
 {
-  val member = HeterogeneousBag(params.members.map(p => new ClockBundle(p)))
+  val member: RecordListMap[ClockBundle] = {
+    val nameToBundleMap = ListMap(params.members.map{p => p.name -> new ClockBundle(p)}: _*)
```

Comment:
As i mentioned below, I think you could make `ClockGroupBundleParameters.members` itself be a `Map[String, ClockBundleParams] and then just iterate through it, turning each item in `values` into a `ClockBundle` here

Reply:
thanks, have done that